app/models/user.rb

```diff
@@ -5,6 +5,50 @@ class User < ApplicationRecord
   include Searchable
   include Storext.model
 
+  # NOTE: @citizen428 This is temporary code during profile migration and will
+  # be removed.
+  # rubocop:disable Metrics/BlockLength
+  concerning :ProfileMigration do
+    included do
+      PROFIE_FIELDS =
+        (Profiles::ExtractData::DIRECT_ATTRIBUTES + Profiles::ExtractData::MAPPED_ATTRIBUTES.values).freeze
+
+      # All new users should automatically have a profile
+      after_create_commit { Profile.create(user: self, data: Profiles::ExtractData.call(self)) }
+
+      # Keep saving changes locally for the time being, but propagate them to profiles.
+      after_update_commit do
+        if (previous_changes.keys.map(&:to_sym) & User::PROFIE_FIELDS).present?
+          profile.update(data: Profiles::ExtractData.call(self))
+        end
+      end
+
+      # Define wrapped getters for profile attributes. We first try to get the
+      # value from the profile and if it doesn't exist there we move retrieve it
+      # from here.
+      Profiles::ExtractData::DIRECT_ATTRIBUTES.each do |attribute|
```

Comment:
These getters allow us to run the profile backfilling at our convenience without any rush.